### PR TITLE
net: openthread: increase openthread stack sizes

### PR DESCRIPTION
**Pull request by pszkotak**
This PR increases the OpenThread stacks
(openthread and openthread radio workqueue)
by 96b for all platforms.

Signed-off-by: Piotr Szkotak <piotr.szkotak@nordicsemi.no>

### CODE REVIEW

**Comment by MarekPorwisz** [comment]
What about shell stack for our samples?

**Comment by edmont** [comment]
How about adding the reason for 96 B to the commit message?

**Comment by pszkotak** [comment]
The sysworkq size is already set to 4k, prefer not to increase it any further.

File: samples/openthread/common/overlay-ot-defaults.conf [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
-CONFIG_NCS_SAMPLES_DEFAULTS=y
+CONFIG_NCS_SAMPLES_DEFAULTS=n
```
Comment (edmont) [inline samples/openthread/common/overlay-ot-defaults.conf:7]: Why this change?

**Review by edmont — APPROVED** [review]

File: samples/openthread/common/overlay-ot-defaults.conf [hunk 2]
```diff
@@ -20,6 +20,9 @@ CONFIG_NET_CONNECTION_MANAGER=y
 CONFIG_MAIN_STACK_SIZE=2048
 CONFIG_INIT_STACKS=y
 
+# Shell stack
+CONFIG_SHELL_STACK_SIZE=3168
```
Comment (MarekPorwisz) [inline samples/openthread/common/overlay-ot-defaults.conf:24]: does it compile with no Kconfig warning/error on NCP sample?

Comment (pszkotak) [reply samples/openthread/common/overlay-ot-defaults.conf]: Good point. In that case we need to introduce the change to zephyr. Working on that.

Comment (pszkotak) [reply samples/openthread/common/overlay-ot-defaults.conf]: https://github.com/zephyrproject-rtos/zephyr/pull/34509

**Comment by pszkotak** [comment]
> The sysworkq size is already set to 4k, prefer not to increase it any further.

It turns out it is not the case - 1024 was applied here, incremented by 96.

File: samples/openthread/common/overlay-ot-defaults.conf [hunk 3]
```diff
@@ -34,6 +34,7 @@ CONFIG_NET_IPV6_MLD=n
 # Stack sizes configuration
 CONFIG_NET_TX_STACK_SIZE=1200
 CONFIG_NET_RX_STACK_SIZE=1500
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=1120
```
Comment (MarekPorwisz) [inline samples/openthread/common/overlay-ot-defaults.conf:37]: Does it work with multiprotocol? I think at least 2KB were required if BT is in use.

config SYSTEM_WORKQUEUE_STACK_SIZE
	int
	default 2048 if BT_LL_SOFTDEVICE

Comment (pszkotak) [reply samples/openthread/common/overlay-ot-defaults.conf:37]: For the CLI I see only the config with the minimal multiprotocol configuration, which has the `CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=1024`. Changing to 2144.

**Review by lmaciejonczyk — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by MarekPorwisz — APPROVED** [review]
